clients/wallet/src/task.rs

```diff
@@ -0,0 +1,206 @@
+use crate::Ledger;
+use tokio::sync::oneshot::{channel, error::TryRecvError, Receiver, Sender};
+
+pub const LEDGER_TASK_MAX_RETRIES: u8 = 3;
```

Comment:
We should consider passing this constant to the constructor function instead of always using it

Reply:
i think it doesn't hurt to have a default value, so I changed it to  `DEFAULT_MAX_RETRIES` and moved it inside the struct itself, to avoid confusion. 
However, i did add a new method: `fn new_with_max_retries(...)` with 3 params; the last param being the `max_retries`

Reply:
That's a great idea, thank you 👍